palace/utils/iodata.cpp

```diff
@@ -16,14 +20,131 @@
 namespace palace
 {
 
+namespace
+{
+
+std::stringstream PreprocessFile(const char *filename)
+{
+  // Read configuration file into memory and return as a stringstream.
+  std::string file;
+  {
+    std::ifstream fi(filename);
+    std::stringstream buf;
+    if (!fi.is_open())
+    {
+      MFEM_ABORT("Unable to open configuration file \"" << filename << "\"!");
+    }
+    buf << fi.rdbuf();
+    fi.close();
+    file = buf.str();
+  }
+
+  // Strip C and C++ style comments (//, /* */) using regex. Correctly handles comments
+  // within strings and escaped comment markers (see tinyurl.com/2s3n8dkr).
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|(\/\*([^*]|[\r\n]|(\*+([^*\/]|[\r\n])))*\*+\/))"
+                   R"(|(\/\/.*))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Also strip whitespace.
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|(\s+))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Also strip erroneous trailing commas.
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|,+(?=\s*?[\}\]]))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Perform integer range expansion for arrays ([a - b, c] = [a-b,c] =
+  // [a,a+1,...,b-1,b,c]). The whole file is now one line and arrays have no spaces after
+  // whitespace stripping.
+  std::stringstream output;
+  auto RangeExpand = [](std::string_view str) -> std::string
+  {
+    // Handle the given string which is only numeric with possible hyphens.
+    int num;
+    auto [ptr, ec] = std::from_chars(str.data(), str.data() + str.length(), num);
+    MFEM_VERIFY(ptr != str.data(), "Invalid integer conversion in range expansion!");
+    if (ptr == str.data() + str.length())
+    {
+      return std::string(str);
+    }
+    // Range specified, expand the bounds.
+    int num2;
+    auto [ptr2, ec2] = std::from_chars(ptr + 1, str.data() + str.length(), num2);
+    MFEM_VERIFY(ptr2 != ptr + 1, "Invalid integer conversion in range expansion!");
```

Comment:
Add similar checks on `ec2` from above here, e.g. if this were `1-5000000000` I think this would bug in a weird way.

Reply:
Fixed in [1325511](https://github.com/awslabs/palace/pull/76/commits/13255116fa34de49bb39efbb4cd74c35c0c7a95f).